core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -3177,7 +3177,11 @@ public PhysicalOperation visitRefreshMaterializedView(RefreshMaterializedViewNod
         public PhysicalOperation visitTableWriter(TableWriterNode node, LocalExecutionPlanContext context)
```

Comment:
commit message could be `Always use preferred insert partitioning for local scaled exchanges`

You could add description that with local scaled exchanges there it no issue with skewed partitions, therefore we can always use preferred insert partitioning when local scaling is enabled